src/arch/src/x86_64/mod.rs

```diff
@@ -163,9 +161,6 @@ pub fn configure_system(
     }
 
     let zero_page_addr = GuestAddress(layout::ZERO_PAGE_START);
-    guest_mem
-        .checked_offset(zero_page_addr, mem::size_of::<boot_params>())
-        .ok_or(Error::ZeroPagePastRamEnd)?;
```

Comment:
Does this need to be removed ?

Reply:
It is redundant since the next line does a call to `guest_mem.write_obj` which checks if [`zero_page_addr` , `zero_page_addr + size_of(params) - 1`] is inside a mapped memory region.

Reply:
Then I think you can also remove `ZeroPagePastRamEnd` error.

Reply:
Done. Please take a look at the last force push.